sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UseConstantsWhereAppropriate.cs

```diff
@@ -97,6 +97,7 @@ private static bool IsFieldRelevant(IFieldSymbol fieldSymbol)
             return fieldSymbol != null &&
                    fieldSymbol.IsStatic &&
                    fieldSymbol.IsReadOnly &&
+                   fieldSymbol.GetEffectiveAccessibility() != Accessibility.Public &&
```

Comment:
I suggest to use `fieldSymbol.IsPubliclyAccessible()` here instead because I assume the problem is the same for `protected` constants.